sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/regexpExpressions.scala

```diff
@@ -232,30 +232,41 @@ case class RLike(left: Expression, right: Expression) extends StringRegexExpress
  * Splits str around pat (pattern is a regular expression).
  */
 @ExpressionDescription(
-  usage = "_FUNC_(str, regex) - Splits `str` around occurrences that match `regex`.",
+  usage = "_FUNC_(str, regex, limit) - Splits `str` around occurrences that match `regex`." +
+    "The `limit` parameter controls the number of times the pattern is applied and " +
+    "therefore affects the length of the resulting array. If the limit n is " +
+    "greater than zero then the pattern will be applied at most n - 1 times, " +
+    "the array's length will be no greater than n, and the array's last entry " +
+    "will contain all input beyond the last matched delimiter. If n is " +
+    "non-positive then the pattern will be applied as many times as " +
+    "possible and the array can have any length. If n is zero then the " +
+    "pattern will be applied as many times as possible, the array can " +
+    "have any length, and trailing empty strings will be discarded.",
```

Comment:
hmm, is it possible to make this usage more compact? I think the usage here should be concise.

Reply:
@viirya i'll take a crack at it -- the usage is a bit funky given the different behavior based on what `limit` is, I aired on the side of verbose